pkg/chunk/metrics.go

```diff
@@ -25,6 +29,7 @@ type cacheManagerMetrics struct {
 	cacheEvicts     prometheus.Counter
 	cacheWriteBytes prometheus.Counter
 	cacheWriteHist  prometheus.Histogram
+	stagingBlocks   atomic.Int64
```

Comment:
Better use a type in prometheus as well.

Reply:
Yes, any suggestions where to put this metric? I didn't find other good places in the codebase, other than a global variable.

Reply:
yeah, looks like we have to use a global variable.

Reply:
If you prefer that, I can modify the code

Reply:
do it please.

Reply:
OK, updated and tested.